### ISSUE TITLE
Double content in notifications on Pebble Time

### ISSUE THREAD

**Issue by nikhau** [smokku/pebble#87]
When receiving sms or email notifications on my Pebble Time Steel, I get the text content displayed/rendered twice within the same single notification.
The first occurence seems limited to a certain amount of characters, the text get cut if too long (don't know the limit).
The second occurence shows entire content.
So when receiving a too long message, I have to scroll passed the first occurence and to the end of the second occurence to be albe to read the entire notification.

Pebble FW 3.4
Sailfish 1.1.9.28
Pebbled 1.4-1

**Comment by RobertMe** [comment]
For SMS this is the expected behavior. The first text is the subject, which I implemented as using the contents (with a max of 64 chars), the second is the body and can be longer. But when dropping the subject you're forced to open the notification.

For email there indeed seems to be an issue. As it should use both the subject and contents of the email. But uses the content for both the notification subject as body.

**Comment by Astaoth** [comment]
On my PTS, the e-mail subject is used for the notification subject and body, instead of the e-mail content.

**Comment by RobertMe** [comment]
Right, found the issue. The parts of the code which I changed to get the notifications showing looks like there actually is a sender, subject, and body available. While there actually is only the name of the sender and the subject available.

But this is a limitation in the way the notifications are being send by SFOS. As the notifications "preview-summary" is the name of the sender, and the notifications "preview-body" is the subject, but there is no way to get the (partial) contents of the email. So it actually is (more or less) impossible to get both the subject and body shown on the Pebble.

And after checking the code again it indeed is possible for the notification to show either the SMS' content or emails subject twice. This is something which can/should be fixed.
